### PR TITLE
Printing colorful EventPrediction results - and more!

### PR DESCRIPTION
**Pull request by TipicoDev**
# Description

Added colorama to the requirements and made a simple code to print win results in green and lose results in red.

# Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been updated in requirements.txt

### CODE REVIEW

**Comment by Tkd-Alex** [comment]
This PR will close https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/issues/45
_____
Moreover, I want to suggest some changes:
- Create a boolean value inside `LoggerSettings`, so the use can choose if want/or not the colors.
- Edit our `EmojiFormatter` and rename to `GlobalFormatter`, `GeneralFormatter`, `Formatter` or whatever name you think can be correct
https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/blob/e944ee9df50008bf5e37c17c29eb103183cd98fc/TwitchChannelPointsMiner/logger.py#L12-L39
- You can take some good code from here:
https://github.com/GramAddict/bot/blob/e84fdee2c5f6991e9e9cd029b687f5ffac219877/GramAddict/core/log.py#L9-L29
With these changes we are able to use the `extra` attribute (as we already do with emoji) and set the colors.
```python
from colorama import Fore
logger.info("Text", extra={"color": f"{Fore.GREEN}"})
```

**Comment by TipicoDev** [comment]
@Tkd-Alex tried my best since I'm not that good on python. Also updated the documentation, but feel free to edit it as you want :)
Please check if that is cool.

**Comment by Tkd-Alex** [comment]
I'll see tomorrow 😁😁

**Comment by Tkd-Alex** [comment]
Sorry today I can't test :disappointed: 
I'll open a very fast review with 2 question.

Run also `pre-commit ` for fix the code-checker.
https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/blob/master/CONTRIBUTING.md#python-styleguide

File: TwitchChannelPointsMiner/TwitchChannelPointsMiner.py [hunk 1]
```diff
@@ -28,6 +28,9 @@
     internet_connection_available,
     set_default_settings,
 )
+from colorama import init
+
+init()
```
Comment (Tkd-Alex) [inline TwitchChannelPointsMiner/TwitchChannelPointsMiner.py:33]: Check If this two line can be moved in `configure_loggers` (after method def)
https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/blob/master/TwitchChannelPointsMiner/logger.py#L58

So we can `init()` only if `settings.colored is True`

File: TwitchChannelPointsMiner/logger.py [hunk 2]
```diff
@@ -85,10 +93,11 @@ def configure_loggers(username, settings):
         )
         file_handler = logging.FileHandler(logs_file, "w", "utf-8")
         file_handler.setFormatter(
-            EmojiFormatter(
+            GlobalFormatter(
```
Comment (Tkd-Alex) [inline TwitchChannelPointsMiner/logger.py:96]: This is just a personal curiosity. 
Have you tested if the file logger works? I'm just scared to have strange "string/chars" in the file logs :thinking:

**Review by Tkd-Alex — CHANGES_REQUESTED** [review]

**Comment by Tkd-Alex** [comment]
For the rest: awesome work! And thanks for your time :partying_face:

File: TwitchChannelPointsMiner/logger.py (hunk 2 above)
Comment (TipicoDev) [reply TwitchChannelPointsMiner/logger.py:96]: I didn't really, but you're actually right. Here's an example:
21/02/21 18:33:33 - INFO - TwitchChannelPointsMiner.classes.WebSocketsPool - [on_message]: [31m[31mEventPrediction: mount (6.77k points) - ele passa do round 100? - Result: LOSE, Gained: -1.01k[0m[0m

File: TwitchChannelPointsMiner/TwitchChannelPointsMiner.py (hunk 1 above)
Comment (Tkd-Alex) [reply TwitchChannelPointsMiner/TwitchChannelPointsMiner.py:33]: I've a little free time slot :)
I will do it

**Review by Tkd-Alex — APPROVED** [review]

**Comment by Yishaqel** [comment]
For me it works perfectly, I would like to suggest a different color for when a bet starts (for example blue color). Thanks for your time in improving this!

**Comment by Tkd-Alex** [comment]
> For me it works perfectly, I would like to suggest a different color for when a bet starts (for example blue color). Thanks for your time in improving this!

@Yishaqel For which message in particular? 
- Place the bet after...
- Going to complete...
- Place 2345 channel...
All of this? 

@TipicoDev do you want to do it? :smile_cat:

**Comment by Yishaqel** [comment]
> > For me it works perfectly, I would like to suggest a different color for when a bet starts (for example blue color). Thanks for your time in improving this!
> 
> @Yishaqel For which message in particular?
> 
> * Place the bet after...
> * Going to complete...
> * Place 2345 channel...
>   All of this?
> 
> @TipicoDev do you want to do it? 😸

In general, messages containing the bet. Anyway I have no problem if you think of doing it differently, it is just good to know who you are betting on.

**Comment by kelukelol** [comment]
`22/02/21 23:22:31 - INFO - [__print_report]: Duration 1:19:38.750500`
No issues have been found throughout the running of the script. 
Works just perfectly.

**Comment by kelukelol** [comment]
`23/02/21 06:50:48 - INFO - [__print_report]: Duration 7:26:40.923265`
It still works, without any troubles. A perfect alternative for Windows users who cant use emojis when running this script.


Not gonna lie, would suggest adding more customization (Which colors should it use for bet win/lose)
And, maybe even color for a message with points being gained. 

For example:
`23/02/21 05:43:50 - INFO - [on_message]: +50 --> Streamer(username=nightbotuhc, channel_id=62713040, channel_points=15.31k) - Reason: CLAIM.` (but in whatever color it's being choosed in the run.py file.)
Again, this is just a suggestion. I am not forcing you to do it. It's just an idea

**Comment by Tkd-Alex** [comment]
Guys for me It's a pleasure that participate actively on this project with PR / Issue and idea :cupid: 
Personally I don't need the coloured terminal, but I see for most of you and especially for Windows user is a great feature! 

We have already discussed here https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/issues/45 how can be "difficult" to find a good colour palette. I've 3 ideas:
 1. Create a customizable theme
 2. Give the user to create a custom theme / colour palette
 3. Make the colours static and If you don't like, It's not my problem :rofl:  

I want just to say if @TipicoDev have the time to work on It and If someone can help to find the "right colours" for each message. So we have a unique PR. 
If this is not the case I'll merge this PR and I hope another user in the future will implement the other message's colour. Currently, I don't have too much time, but maybe, I can also work on this in the future :smile_cat:

**Comment by TipicoDev** [comment]
Sorry for the delay. I've been kinda busy at work this days.
@Tkd-Alex I don't really have much time to work on it for now, I guess we can just merge this as a WIP and improve it whenever me or someone else have some time.
Thanks a lot for everyone's help! =)

File: TwitchChannelPointsMiner/TwitchChannelPointsMiner.py (hunk 1 above)
Comment (TipicoDev) [reply TwitchChannelPointsMiner/TwitchChannelPointsMiner.py]: Fixed it on the last commit!

**Comment by Tkd-Alex** [comment]
Someone can test my last commit? :partying_face: 
Look at the README.md before the example.
```python
from colorama import Fore
ColorPalette(
    "STREAMER_ONLINE" = Fore.GREEN,
    "STREAMER_OFFLINE" = Fore.RED,
    "GAIN_FOR_RAID" = Fore.YELLOW,
    "GAIN_FOR_CLAIM" = Fore.YELLOW,
    "GAIN_FOR_WATCH" = Fore.YELLOW,
    "BET_WIN" = Fore.GREEN,
    "BET_LOSE" = Fore.RED,
    "BET_REFUND" = Fore.RESET,
    "BET_FILTERS" = Fore.MAGENTA,
    "BET_GENERAL" = Fore.BLUE,
    "BET_FAILED" = Fore.RED,
)
```

**Comment by Yishaqel** [comment]
This is working correctly but I recommend to add the possibility to add a color to:

26/02 11:09:01 - Place the bet after: 55.39s for: EventPrediction: Streamer (333.33k points) - Top 2

**Comment by Tkd-Alex** [comment]
> This is working correctly but I recommend to add the possibility to add a color to:
> 
> 26/02 11:09:01 - Place the bet after: 55.39s for: EventPrediction: Streamer (333.33k points) - Top 2

In `BET_GENERAL` or in separated key?

**Comment by Yishaqel** [comment]
> > This is working correctly but I recommend to add the possibility to add a color to:
> > 26/02 11:09:01 - Place the bet after: 55.39s for: EventPrediction: Streamer (333.33k points) - Top 2
> 
> In `BET_GENERAL` or in separated key?

For me it would be better in a different key to recognize who is starting a bet

**Pull request merged**